File: AngularUI/UI/src/app/core/services/translate.service.ts
import { TranslateModel } from './../models/translate/translateModel';
import { Service } from './service';
import { LocalStorageService } from './local-storage.service';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TranslateAddModel } from '../models/translate/translateAddModel';
import { Observable } from 'rxjs';
import { DeleteModel } from '../models/deleteModel';
import { ListResponseModel } from '../models/response/listResponseModel';
import { ResponseModel } from '../models/response/responseModel';
import { SingleResponseModel } from '../models/response/singleResponseModel';
import { apiUrl } from 'src/api';

@Injectable({
  providedIn: 'root',
})
export class TranslateService
  implements Service<TranslateAddModel, TranslateModel>
{
  constructor(
    private httpClient: HttpClient,
    private localStorageService: LocalStorageService
  ) {}

  add(addModel: TranslateAddModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/add';
    return this.httpClient.post<ResponseModel>(newPath, addModel);
  }

  delete(deleteModel: DeleteModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/delete';
    return this.httpClient.post<ResponseModel>(newPath, deleteModel);
  }

  update(updateModel: TranslateModel): Observable<ResponseModel> {
    let newPath = apiUrl + 'translates/update';
    return this.httpClient.post<ResponseModel>(newPath, updateModel);
  }

  getById(id: number): Observable<SingleResponseModel<TranslateModel>> {
    let newPath = apiUrl + 'translates/getbyid?id=' + id;
    return this.httpClient.get<SingleResponseModel<TranslateModel>>(newPath);
  }

  getAll(): Observable<ListResponseModel<TranslateModel>> {
    let newPath = apiUrl + 'translates/getall';
    return this.httpClient.get<ListResponseModel<TranslateModel>>(newPath);
  }
}

File: FlutterUI/flutter_ui/lib/core/services/translate_service.dart
import 'package:flutter_ui/core/models/delete_model.dart';
import 'package:flutter_ui/core/models/response/list_response_model.dart';
import 'package:flutter_ui/core/models/response/response_model.dart';
import 'package:flutter_ui/core/models/response/single_response_model.dart';
import 'package:flutter_ui/core/models/translate/translate_add_model.dart';
import 'package:flutter_ui/core/models/translate/translate_model.dart';
import 'package:flutter_ui/core/utilities/dependency_resolver.dart';
import 'package:flutter_ui/core/utilities/service_repository.dart';

class TranslateService extends ServiceRepositoryWithAddModel<TranslateAddModel,
    TranslateModel, int> {
  @override
  Future<ResponseModel> add(TranslateAddModel addModel) async {
    var response = await httpClient.post(
      "translates/add",
      body: addModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ResponseModel> delete(DeleteModel deleteModel) async {
    var response = await httpClient.post(
      "translates/delete",
      body: deleteModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  @override
  Future<ListResponseModel<TranslateModel>> getAll() async {
    var response = await httpClient.get("translates/getall");

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  @override
  Future<SingleResponseModel<TranslateModel>> getById(int id) async {
    var response = await httpClient.get(
      "translates/getbyid",
      queryParameters: {"id": id},
    );

    return SingleResponseModel<TranslateModel>.fromJson(response);
  }

  @override
  Future<ResponseModel> update(TranslateModel updateModel) async {
    var response = await httpClient.post(
      "translates/update",
      body: updateModel.toJson(),
    );

    return ResponseModel.fromJson(response);
  }

  Future<ListResponseModel<TranslateModel>> getByKey(String key) async {
    var response = await httpClient.get(
      "translates/getbykey",
      queryParameters: {"key": key},
    );

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  Future<ListResponseModel<TranslateModel>> getByLanguageId(
      int languageId) async {
    var response = await httpClient.get(
      "translates/getbylanguageid",
      queryParameters: {"languageId": languageId},
    );

    return ListResponseModel<TranslateModel>.fromJson(response);
  }

  Future<Map?> getTranslates(String? languageCode) async {
    var response = await httpClient.get(
      "translates/gettranslates",
      queryParameters: {"languageCode": languageCode},
    );

    return response?.data;
  }
}
